### PR TITLE
Add Alibaba Event Collector

### PR DESCRIPTION
**Pull request by adi88d**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
- Add params to IntegrationHTTPRequest
- Add `Alibaba Action Trail` pack with the integration `Alibaba Action Trail Event Collector`

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/AlibabaActionTrail/Integrations/AlibabaActionTrailEventCollector/AlibabaActionTrailEventCollector.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+def get_alibaba_timestamp_format(value):
+    timestamp: Optional[datetime]
+    if isinstance(value, int):
+        return value
+    if not isinstance(value, datetime):
+        timestamp = dateparser.parse(value)
+    if timestamp is None:
+        raise TypeError(f'after is not a valid time {value}')
+    return int(time.mktime(timestamp.timetuple()))
```
Comment (yaakovi) [inline Packs/AlibabaActionTrail/Integrations/AlibabaActionTrailEventCollector/AlibabaActionTrailEventCollector.py:134]: can `value` be a `datetime` object?
If so, then the last return will raise an attribute error (NoneType has no attribute `timetuple`)
If not, then the second condition is redundant

File: Packs/AlibabaActionTrail/Integrations/AlibabaActionTrailEventCollector/AlibabaActionTrailEventCollector.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        content = content.encode('utf-8')
+    if isinstance(key, six.text_type):  # hmac.new accept 8-bit str
+        key = key.encode('utf-8')
+
+    hashed = hmac.new(key, content, hashlib.sha1).digest()
+    return base64_encodestring(hashed).rstrip()
+
+
+def get_request_authorization(resource, key, req_params, req_headers):
+    """ :return bytes (PY2) or string (PY2) """
```
Comment (yaakovi) [inline Packs/AlibabaActionTrail/Integrations/AlibabaActionTrailEventCollector/AlibabaActionTrailEventCollector.py:118]: ?

File: Packs/AlibabaActionTrail/Integrations/AlibabaActionTrailEventCollector/AlibabaActionTrailEventCollector.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+
+def canonicalized_resource(resource, params):
+    if params:
+        urlString = ''
+        for key, value in sorted(six.iteritems(params)):
+            urlString += u"{0}={1}&".format(
+                key, value.decode('utf8') if isinstance(value, six.binary_type) else value)
+        resource += '?' + urlString[:-1]
+        if six.PY3:
```
Comment (yaakovi) [inline Packs/AlibabaActionTrail/Integrations/AlibabaActionTrailEventCollector/AlibabaActionTrailEventCollector.py:90]: The integration will always run on python3.
Why do we need this condition?

File: Packs/AlibabaActionTrail/Integrations/AlibabaActionTrailEventCollector/AlibabaActionTrailEventCollector.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+            return resource
+        else:
+            return resource.encode('utf8')
+
+    return resource
+
+
+def base64_encodestring(s):
+    if six.PY2:
```
Comment (yaakovi) [inline Packs/AlibabaActionTrail/Integrations/AlibabaActionTrailEventCollector/AlibabaActionTrailEventCollector.py:99]: Same here

File: Packs/AlibabaActionTrail/Integrations/AlibabaActionTrailEventCollector/AlibabaActionTrailEventCollector.py (hunk 1 above)
Comment (adi88d) [reply Packs/AlibabaActionTrail/Integrations/AlibabaActionTrailEventCollector/AlibabaActionTrailEventCollector.py:134]: the value can be a string ISO format when execute `alibaba-get-events` command or `time interval` like 3 days from the integration params or Unix Timestamp (int) when get it `from getLatsRun`.  
removing the second condition

**Review by yaakovi — APPROVED** [review]

**Pull request merged**